lib/Doctrine/Migrations/Tools/Console/ConsoleInputMigratorConfigurationFactory.php

```diff
@@ -31,12 +32,26 @@ public function getMigratorConfiguration(InputInterface $input): MigratorConfigu
 
     private function determineAllOrNothingValueFrom(InputInterface $input): ?bool
     {
-        $allOrNothingOption = null;
+        $allOrNothingOption        = null;
+        $wasOptionExplicitlyPassed = $input->hasOption('all-or-nothing');
 
-        if ($input->hasOption('all-or-nothing')) {
+        if ($wasOptionExplicitlyPassed) {
             $allOrNothingOption = $input->getOption('all-or-nothing');
         }
 
+        if ($wasOptionExplicitlyPassed && $allOrNothingOption !== null) {
+            Deprecation::trigger(
+                'doctrine/migrations',
+                'https://github.com/doctrine/migrations/issues/1304',
+                <<<'DEPRECATION'
+                    Context: Passing values to the `--all-or-nothing`
+                    Problem: The mechanism to read the option value is unnecessarily complicated.
```

Comment:
The problem must describe what the user is doing wrong, not why we decided to deprecate passing values.

```suggestion
                    Problem: Passing values is deprecated
```